File: app/src/main/java/cc/aoeiuv020/demo/MainActivity.java
package cc.aoeiuv020.demo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.tvHello).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Log.d(TAG, "onClick() called with: v = [" + v + "]");
                Reporter.getInstance().post("click,");
            }
        });
        findViewById(R.id.tvHello).setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View v) {
                Log.d(TAG, "onLongClick() called with: v = [" + v + "]");
                Reporter.getInstance().post("long click,");
                throw new IllegalStateException("tvHello long click,");
            }
        });
    }
}

File: app/src/main/java/cc/aoeiuv020/demo/MainActivity.kt
package cc.aoeiuv020.demo

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnListView.setOnClickListener {
            ListViewActivity.start(this)
        }

        btnNested.setOnClickListener {
            NestedActivity.start(this)
        }

        btnGrid.setOnClickListener {
            GridActivity.start(this)
        }

        btnPagerGrid.setOnClickListener {
            PagerGridActivity.start(this)
        }

        btnCoordinator.setOnClickListener {
            CoordinatorActivity.start(this)
        }

        btnGridItemHeight.setOnClickListener {
            GridItemHeightActivity.start(this)
        }
    }
}
